### PR TITLE
[IMP] fleet: Add possibility to mass mail drivers

### PR DESCRIPTION
**Pull request by jugj-odoo**
task-3459884

### CODE REVIEW

File: addons/fleet/wizard/fleet_vehicle_send_mail.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    def action_send(self):
+        self.ensure_one()
+        without_emails = self.vehicle_ids.driver_id.filtered(lambda a: not a.email)
+        if without_emails:
+            return {
+                'type': 'ir.actions.client',
+                'tag': 'display_notification',
+                'params': {
+                    'type': 'danger',
+                    'message': _("The following vehicle drivers are missing an email address: %s.", ', '.join(without_emails.mapped(lambda a: a.name))),
```
Comment (leonardopavanrocha) [inline addons/fleet/wizard/fleet_vehicle_send_mail.py:31]: mapped accepts either a lambda function or a string with the name of the field you want to get. In this case, it's easier to just pass "name"
```suggestion
                    'message': _("The following vehicle drivers are missing an email address: %s.", ', '.join(without_emails.mapped("name"))),
```

File: addons/fleet/views/mail_compose_views.xml
Comment (leonardopavanrocha) [inline addons/fleet/views/mail_compose_views.xml:1]: empty file :eyes:

File: addons/fleet/wizard/fleet_vehicle_send_mail.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+            record.write({'template_id': template.id})
+
+            return {
+                'type': 'ir.actions.act_window',
+                'view_mode': 'form',
+                'res_id': template.id,
+                'res_model': 'mail.template',
+                'target': 'new',
+            }
```
Comment (leonardopavanrocha) [inline addons/fleet/wizard/fleet_vehicle_send_mail.py:71]: This seems inconsistent. From your `for record in self`, I assume this method can manipulate multiple wizards (which I don't think is the case). But then, you return on the first iteration, so processing a single record. I guess you can just drop the for loop and it should be good

**Review by leonardopavanrocha — CHANGES_REQUESTED** [review]
Thanks for working on this! I have some small comments. Let me know if anything is not clear :slightly_smiling_face:

**Comment by jugj-odoo** [comment]
@leonardopavanrocha changes applied

File: addons/fleet/wizard/fleet_vehicle_send_mail.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    vehicle_ids = fields.Many2many('fleet.vehicle', string='Vehicles', required=True)
+    author_id = fields.Many2one('res.partner', 'Author', required=True, default=lambda self: self.env.user.partner_id.id)
+    template_id = fields.Many2one(domain=lambda self: [('model_id', '=', self.env['ir.model']._get('fleet.vehicle').id)])
+    attachment_ids = fields.Many2many(
+        'ir.attachment', 'fleet_vehicle_mail_compose_message_ir_attachments_rel',
+        'wizard_id', 'attachment_id', string='Attachments')
+
+    @api.onchange('template_id')
+    def _onchange_template_id(self):
+        self.attachment_ids = self.template_id.attachment_ids
```
Comment (leonardopavanrocha) [inline addons/fleet/wizard/fleet_vehicle_send_mail.py:20]: maybe attachment_ids could be a related field on `template_id.attachment_ids`

**Review by leonardopavanrocha — COMMENTED** [review]
Just one final suggestion. Other than that LGTM!

File: addons/fleet/wizard/fleet_vehicle_send_mail.py (hunk 3 above)
Comment (leonardopavanrocha) [reply addons/fleet/wizard/fleet_vehicle_send_mail.py:20]: Related fields are readonly by default, so it wouldn't change the original template AFAIK

Comment (jugj-odoo) [reply addons/fleet/wizard/fleet_vehicle_send_mail.py:20]: but then we can't add or remove attachments to our mail :D

Comment (leonardopavanrocha) [reply addons/fleet/wizard/fleet_vehicle_send_mail.py:20]: ah :facepalm:

**Review by leonardopavanrocha — APPROVED** [review]
LGTM!

**Review by Bertrand2 — APPROVED** [review]

**Comment by Bertrand2** [comment]
robodoo r+

**Pull request closed without merging**